### PR TITLE
ci: version bump for ghcr.io/spack/e4s-amazonlinux-2

### PR DESCRIPTION
**Pull request by zackgalbreath**
This new image comes with GnuPG v2.4.0

### CODE REVIEW

**Review by haampie — DISMISSED** [review]

**Review by haampie — DISMISSED** [review]
Self-approving, (sorry @adamjstewart), since rn tons of PRs are blocked by failing py-gevent.

**Comment by adamjstewart** [comment]
How did the failure start? Why haven't we seen this until now? I still think we should limit which versions of python py-gevent supports or manually delete cythonized files. But I'm fine with doing that in a follow-up PR (I trust @haampie)

**Comment by haampie** [comment]
I don't even know how to explain this in a one sentence, but basically we were
sort of lucky that cython previously erroneously recythonized py-gevent.

py-gevent includes generated (but broken) C-files. Normally it should *not*
regenerate those, but it did, because dumb cython regenerates cython files
if *its own* pyx files have newer mtime than the source files of the thing that's
being installed. :sob: 

So, what changed was #35623 which sets mtimes of binary packages to 0 so that
repeated invocations of `spack buildcache create` produces bitwise equal tarballs.

In short: cython stopped incorrectly regenerating C-files, which revealed that
the pre-generated C-files were broken. Or shorter: two bugs cancel each other
out.

**Comment by haampie** [comment]
3 tensorflows and 3 paraviews to go...

**Comment by adamjstewart** [comment]
Gotta get that `-j` working. Want me to submit a PR?

**Comment by haampie** [comment]
I think the `-j` is fine for the tensorflow, just not for jaxlib

**Comment by adamjstewart** [comment]
Ah, I see we don't directly invoke bazel in jaxlib so we'll need to patch things like we did in dm-tree.

**Comment by haampie** [comment]
of course... this needs a rebase :weary: :weary: :weary:  there goes 6 hours of paraview builds right out of the window. Someone should force merge this one...

**Review by haampie — APPROVED** [review]

**Comment by kwryankrattiger** [comment]
Unit tests fixed in #36004
